app/components/e-checkbox.vue

```diff
@@ -0,0 +1,282 @@
+<template>
+  <div :class="b(modifiers)">
+
+    <input
+      :aria-checked="checked ? 'true' : 'false'"
+      :class="b('field')"
+      :disabled="disabled"
+      :value="value"
+      :name="name"
+      :id="name"
+      v-bind="$attrs"
+      v-model="internalValue"
+      role="checkbox"
+      type="checkbox"
+      @blur="onBlur"
+      @change="onChange"
+      @focus="onFocus">
+
+    <label :for="name"
+           :class="b('label')"
+           @mouseenter="isHover = true"
+           @mouseleave="isHover = false">
+      <slot></slot>
+    </label>
+
+  </div>
+
+</template>
+
+<script>
+  import formStates from '@/mixins/form-states';
+
+  export default {
+    name: 'e-checkbox',
+    mixins: [formStates],
+    inheritAttrs: false,
+
+    model: {
+      /**
+       * Changes v-model behavior and use 'checked' instead of 'value' as prop.
+       * Avoids conflict with default value attribute.
+       */
+      prop: 'checked',
+      event: 'change',
+    },
+
+    props: {
+      /**
+       * Adds name attribute
+       * Note: is also used as id and for label (won't work without them)
+       */
+      name: {
+        type: String,
+        required: true,
+      },
+
+      /**
+       * Adds value attribute
+       */
+      value: {
+        type: String,
+        required: true,
+      },
+    },
+
+    // data() {
+    //   return {};
+    // },
+
+    computed: {
+      /**
+       * Sets value of component model to parent model
+       *
+       * @returns  {Boolean}   Status of the checkbox
+       */
+      internalValue: {
+        get() {
+          return this.checked;
+        },
+        set(value) {
+          this.$emit('change', value);
+        }
+      },
+
+      /**
+       * Defines state modifier classes
+       *
+       * @returns  {Object}   BEM classes
+       */
+      modifiers() {
+        return {
+          ...this.stateModifiers
+        };
+      }
+    },
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      /**
+       * Emits state to parent and wrapper component
+       * Update "isChecked" state
+       *
+       * @param   {Boolean}  event   Field input
+       */
+      onChange(event) {
+        this.isChecked = event.target.checked;
+        this.$parent.$emit('change');
+      },
+
+      /**
+       * Emits focus to parent and wrapper component
+       * Update "hasFocus" state
+       */
+      onFocus() {
+        this.hasFocus = true;
+
+        this.$emit('focus');
```

Comment:
If you add a comment directly in the line before the `$emit` it will add the comment to the styleguide (see https://github.com/vue-styleguidist/vue-styleguidist/blob/master/docs/Documenting.md#using-jsdoc-tags)

Reply:
Done for all.